### PR TITLE
backend: remove some clang generated warnings

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by ZombineDev** [comment]
Can you post the warnings you got, just for reference?

**Review by PetarKirov — APPROVED** [review]

File: src/ddmd/backend/gloop.c [hunk 1]
```diff
@@ -2104,8 +2104,7 @@ STATIC void findbasivs(loop *l)
 
         biv->IVbasic = s;               // symbol of basic IV
 
-        cmes3("Symbol '%s' (%d) is a basic IV, ",s->Sident
-                ? (char *)s->Sident : "",i);
+        cmes3("Symbol '%s' (%d) is a basic IV, ", s->Sident, i);
```
Comment (WalterBright) [inline src/ddmd/backend/gloop.c]: It's a static array, it is always non-NULL.

File: src/ddmd/backend/gloop.c [hunk 2]
```diff
@@ -2231,8 +2230,7 @@ STATIC void findopeqs(loop *l)
 
         biv->IVbasic = s;               // symbol of basic IV
 
-        cmes3("Symbol '%s' (%d) is an opeq IV, ",s->Sident
-                ? (char *)s->Sident : "",i);
+        cmes3("Symbol '%s' (%d) is an opeq IV, ",s->Sident,i);
```
Comment (WalterBright) [inline src/ddmd/backend/gloop.c]: The backend doesn't use that style.

File: src/ddmd/backend/gloop.c (hunk 1 above)
Comment (ZombineDev) [inline src/ddmd/backend/gloop.c]: Thanks for the clarification, I didn't see that it was a static array.

**Comment by ZombineDev** [comment]
All checks have passed, so feel free to merge.

**Review by andralex — APPROVED** [review]

**Pull request merged**